locust/test/test_dispatch.py

```diff
@@ -3612,61 +3781,81 @@ class User5(User):
+                                    users_dispatcher.get_current_user_count(user_class.__name__),
                                     user_class.fixed_count,
                                     msg=f"{user_class.__name__}, {target_user_count}",
                                 )
 
                         # Ramp-down to go to `down_to_count`
                         # and ensure the fixed users was decreased too
+                        if self.user_dispatcher_class == FixedUsersDispatcher:
+                            target = {user_class.__name__: 0 for user_class in users_dispatcher._user_classes}
+                        else:
+                            target = down_to_count
 
-                        users_dispatcher.new_dispatch(target_user_count=down_to_count, spawn_rate=1)
+                        users_dispatcher.new_dispatch(target_user_count=target, spawn_rate=1)
                         users_dispatcher._wait_between_dispatch = 0
 
                         list(users_dispatcher)
 
                         for user_class in user_classes:
                             if user_class.fixed_count:
                                 self.assertNotEqual(
-                                    users_dispatcher._get_user_current_count(user_class.__name__),
+                                    users_dispatcher.get_current_user_count(user_class.__name__),
                                     user_class.fixed_count,
                                 )
 
                         # Ramp-up go back to `target_user_count` and ensure
                         # the fixed users return to their counts
-
-                        users_dispatcher.new_dispatch(target_user_count=target_user_count, spawn_rate=1)
+                        if self.user_dispatcher_class == FixedUsersDispatcher:
+                            target = {
+                                user_class.__name__: user_class.fixed_count
+                                for user_class in users_dispatcher._user_classes
+                            }
+                        else:
+                            target = target_user_count
+
+                        users_dispatcher.new_dispatch(target_user_count=target, spawn_rate=1)
                         users_dispatcher._wait_between_dispatch = 0
 
                         list(users_dispatcher)
 
                         for user_class in user_classes:
                             if user_class.fixed_count:
                                 self.assertEqual(
-                                    users_dispatcher._get_user_current_count(user_class.__name__),
+                                    users_dispatcher.get_current_user_count(user_class.__name__),
                                     user_class.fixed_count,
                                 )
 
 
-class TestRampUpDifferentUsers(unittest.TestCase):
-    def test_ramp_up_different_users_for_each_dispatch(self):
+@parameterized_class(PARAMETER_DISPATCHERS)
```

Comment:
I think it makes more sense to duplicate this test rather than parameterizing it. Too many if statements for it to make sense as parameterized.